scripts/etc/init.d/rcS

```diff
@@ -533,66 +533,66 @@ fi
+            echo "OK"
+        fi
+        echo -n "  Setting root password... "
+        echo -n "root:$rootpw" | chroot /rootfs /usr/sbin/chpasswd || fail
         echo "OK"
-    else
-        echo "FAILED !"
     fi
 fi
 
 # add user to system
 if [ ! -z "$username" ]; then
-    echo -n "  Creating user... "
-    chroot /rootfs /usr/sbin/adduser "$username" --gecos "" --disabled-password
-    if [ $? -eq 0 ]; then
+    echo -n "  Creating $username user... "
+    chroot /rootfs /usr/sbin/adduser "$username" --gecos "" --disabled-password || fail
+    echo "OK"
+    # add SSH key for user (if provided)
+    if [ ! -z "$user_ssh_pubkey" ]; then
+        echo -n "  Setting $username SSH key... "
+        ssh_dir="/rootfs/home/$username/.ssh"
+        mkdir -p "$ssh_dir" && \
+            echo "$user_ssh_pubkey" > "$ssh_dir/authorized_keys" || fail
         echo "OK"
-        echo -n "  Setting user password... "
-        echo -n $username:$userpw | chroot /rootfs /usr/sbin/chpasswd
-        if [ $? -eq 0 ]; then
-            echo "OK"
-        else
-            echo "FAILED !"
-        fi
-        if [ ! -z "$user_ssh_pubkey" ]; then
-            echo -n "  Setting user SSH key... "
-            ssh_dir="/home/$username/.ssh"
-            mkdir -p "/rootfs$ssh_dir" && \
-                chmod 700 "/rootfs$ssh_dir" && \
-                echo "$user_ssh_pubkey" > "/rootfs$ssh_dir/authorized_keys"
-            if [ $? -eq 0 ]; then
-                echo "OK"
-                echo -n "  Setting owner on SSH key... "
-                chroot /rootfs /bin/chown -R "$username:$username" "$ssh_dir"
-                if [ $? -eq 0 ]; then
-                    echo "OK"
-                else
-                    echo "FAILED !"
-                fi
-            else
-                echo "FAILED !"
-            fi
-        fi
-        if [ "$user_is_admin" = "1" ]; then
-            echo -n "  Adding $username to sudo group... "
-            chroot /rootfs /usr/sbin/usermod -aG sudo "$username"
+        echo -n "  Setting owner as $username on SSH directory... "
+        chroot /rootfs /bin/chown -R "$username:$username" "/home/$username/.ssh" || fail
+        echo "OK"
+        echo -n "  Setting permissions on $username SSH directory... "
+        chmod -R go-rwx "$ssh_dir" || fail
+        echo "OK"
+    fi
+    if [ ! -z "$user_passwd" ]; then
+        echo -n "  Setting $username password... "
+        echo -n "$username:$userpw" | chroot /rootfs /usr/sbin/chpasswd || fail
+        echo "OK"
+    fi
+    if [ "$user_is_admin" = "1" ]; then
+        echo -n "  Adding $username to sudo group... "
+        chroot /rootfs /usr/sbin/usermod -aG sudo "$username" || fail
+        echo "OK"
+        if [ -z "$user_passwd" ]; then
+            echo -n "  Setting $username to sudo without a password... "
+            echo -n "%sudo         ALL = (ALL) NOPASSWD: ALL" > /rootfs/etc/sudoers.d/01_sudo || fail
```

Comment:
So here is my problem.  I need to distribute the bootable unattended remote network installs to people who have a tendency to lose things.  Hard coding a password is a security issue since it can be logged in with via console if the SDcard is lost; hence the PKI using only SSH keys is preferable.

Creating a service account that can only be remotely logged into without a password required to sudo allows for hooking the unattended installer into a CI/CD setup to automate the rest of the installation and configuration.

Following the somewhat standard cloud methods to enable a service account for automation with sudo without compromising security is the best solution I could come up with.